File: src/main/java/edu/marmara/mapper/impl/CourseMapperImpl.java
package edu.marmara.mapper.impl;

import edu.marmara.dto.CourseGetDTO;
import edu.marmara.mapper.CourseMapper;
import edu.marmara.model.Course;
import edu.marmara.model.DayName;
import edu.marmara.model.WeeklyDate;

import java.util.ArrayList;

public class CourseMapperImpl implements CourseMapper {
    @Override
    public Course mapTo(CourseGetDTO courseGetDTO) {
        Course course = new Course();

        course.setCourseCode(courseGetDTO.getCourseCode());
        course.setCourseTitle(courseGetDTO.getCourseTitle());
        course.setGivenSemester(courseGetDTO.getGivenSemester());
        course.setCourseCredit(courseGetDTO.getCourseCredit());

        course.setDates(new ArrayList<>());

        for (String date : courseGetDTO.getWeeklyDate()) {
            String[] dayAndHour = date.split(" ");

            DayName dayName = DayName.valueOf(dayAndHour[0]);
            WeeklyDate weeklyDate = new WeeklyDate();
            weeklyDate.setDayName(dayName);
            weeklyDate.setHours(Integer.valueOf(dayAndHour[1]));

            course.getDates().add(weeklyDate);
        }

        course.setMaxSeats(courseGetDTO.getAvailableSeats());
        course.setTakenSeats(0);

        return course;
    }
}

File: python/src/main/marmara/mapper/impl/CourseMapperImpl.py
from src.main.marmara.dto import CourseGetDTo
from src.main.marmara.mapper import CourseMapper
from src.main.marmara.model import Course, DayName, WeeklyDate

class CourseMapperImpl(CourseMapper):
    def mapTo(self, courseGetDTO: CourseGetDTo) -> Course:
        course = Course()

        course.courseCode = courseGetDTO.courseCode
        course.courseTitle = courseGetDTO.courseTitle
        course.givenSemester = courseGetDTO.givenSemester
        course.courseCredit = courseGetDTO.courseCredit

        course.dates = []

        for date in courseGetDTO.weeklyDate:
            day_and_hour = date.split(" ")

            day_name = DayName[day_and_hour[0]]
            weekly_date = WeeklyDate()
            weekly_date.dayName = day_name
            weekly_date.hours = int(day_and_hour[1])

            course.dates.append(weekly_date)

        course.maxSeats = courseGetDTO.maxSeats
        course.takenSeats = courseGetDTO.takenSeats
        course.minCreditReq = courseGetDTO.minCreditReq

        return course
